sycl/doc/design/SYCLBINDesign.md

```diff
@@ -15,54 +16,60 @@ The content of the SYCLBIN may be contained as an entry in the offloading binary
 format produced by the clang-offload-packager, as described in
 [ClangOffloadPackager.rst](https://github.com/intel/llvm/blob/sycl/clang/docs/ClangOffloadPackager.rst).
 
-The following illustration gives an overview of how the file format is
-structured.
-
-![SYCLBIN binary file format illustration](SYCLBIN_file_format_illustration.svg)
-
-### Header
-
-The header segment appears as the first part of the SYCLBIN file. Like many
+The format of these files consist of a [file header](#file-header), containing
+general information about the file, followed by three lists of headers: The
+[abstract module header](#abstract-module-header) list, the
+[IR module header](#ir-module-header) list and
+[native device code image header](#native-device-code-image-header) list,
+containing information about the [abstract modules](#abstract-module),
+[IR modules](#ir-module) and
+[native device code images](#native-device-code-image) respectively.
+
+|                                                                       |
+| --------------------------------------------------------------------- |
+| [File header](#file-header)                                           |
+| [Abstract module header](#abstract-module-header) 1                   |
+| ...                                                                   |
+| [Abstract module header](#abstract-module-header) N                   |
+| [IR module header](#ir-module-header) 1                               |
+| ...                                                                   |
+| [IR module header](#ir-module-header) M                               |
+| [Native device code image header](#native-device-code-image-header) 1 |
+| ...                                                                   |
+| [Native device code image header](#native-device-code-image-header) L |
+| Byte table                                                            |
```

Comment:
I think it would be better to have a separate "property set table".  This would still be a blob of bytes, but it will be more efficient if all the property sets are collected together, just because they will be packed together into a small number of pages.

Reply:
I've split them into two byte tables. 👍